sympy/functions/special/zeta_functions.py

```diff
@@ -582,6 +582,40 @@ def _eval_rewrite_as_zeta(self, s, **kwargs):
         return (1 - 2**(1 - s)) * zeta(s)
 
 
+class riemann_xi(Function):
+    r"""
+    Riemann Xi function.
+
+    Examples
+    ========
+
+    The Riemann Xi function is closely related to the Riemann zeta function.
+    The zeros of Riemann Xi function is precisely the non-trivial zeros
+    of the zeta function.
+
+    >>> from sympy import riemann_xi, zeta
+    >>> from sympy.abc import s
+    >>> riemann_xi(s).rewrite(zeta)
+    pi**(-s/2)*s*(s - 1)*gamma(s/2)*zeta(s)/2
+
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Riemann_Xi_function
+
+    """
+
+    from sympy import gamma
+    @classmethod
+    def eval(cls, s):
+        z = zeta(s)
+        if not z.has(zeta):
```

Comment:
Why does `z.has(zeta)` decides the evaluation?

Reply:
Could this be `if not isinstance(z, zeta):`?